### PR TITLE
[geeksandsundry] Add new extractor

### PR DESCRIPTION
**Pull request by ciferkey**
For issue #5946

### CODE REVIEW

File: youtube_dl/extractor/geeksandsundry.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+    def _real_extract(self, url):
+        mobj = re.match(self._VALID_URL, url)
+        page = mobj.group('title')
+        webpage_url = "https://geekandsundry.com/" + page
+        webpage = self._download_webpage(webpage_url, page)
+
+        self.report_extraction(page)
+
+        video_id = self._html_search_regex(r'data-video-id=\"(\d+)\"', webpage, u'video id')
```
Comment (yan12125) [inline youtube_dl/extractor/geeksandsundry.py]: There's no need to use ```u``` prefix given ```unicode_literals``` is declared.

File: youtube_dl/extractor/geeksandsundry.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            'id': 'tabletop-bonus-wils-final-thoughts-on-dread/',
+            'ext': 'mp4',
+            'title': 'TableTop Bonus! Wil\u2019s Final Thoughts on Dread | Geek and Sundry',
+        }
+    }
+
+
+    def _real_extract(self, url):
+        mobj = re.match(self._VALID_URL, url)
+        page = mobj.group('title')
```
Comment (yan12125) [inline youtube_dl/extractor/geeksandsundry.py]: Usually variables named ```page``` are for storing downloaded data (webpage, API page, etc.). This variable is usually named as ```video_id```, ```media_id```, etc. Also, it's better to use ```_match_id``` function here.

File: youtube_dl/extractor/geeksandsundry.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        webpage_url = "https://geekandsundry.com/" + page
+        webpage = self._download_webpage(webpage_url, page)
+
+        self.report_extraction(page)
+
+        video_id = self._html_search_regex(r'data-video-id=\"(\d+)\"', webpage, u'video id')
+        pub_id = self._html_search_regex(r'data-account=\"(\d+)\"', webpage, u'pub id')
+
+        video_url = "http://c.brightcove.com/services/mobile/streaming/index/master.m3u8?videoId=%s&pubId=%s" % (video_id, pub_id)
```
Comment (yan12125) [inline youtube_dl/extractor/geeksandsundry.py]: It's a meta M3U8. It's better to extract formats in it via ```_extract_m3u8_formats``` function.

**Comment by dstftw** [comment]
This is now covered by generic support for new Brightcove embeds.

**Pull request closed without merging**